test/integration/sessions/sessions.spec.test.ts

```diff
@@ -4,7 +4,6 @@ import { runUnifiedSuite } from '../../tools/unified-spec-runner/runner';
 describe('Sessions spec tests', function () {
   runUnifiedSuite(loadSpecTests('sessions'), [
     // TODO(NODE-3951): fix broken dirty sessions spec tests
-    'Dirty explicit session is discarded (insert)',
```

Comment:
This is interesting, the updated test is actually the "implicit session" one, not the explicit session. If the explicit session discard is passing now, perhaps the other one is, too? For unrelated reasons? (maybe the connect work?) Can we try unskipping all the tests mentioned in NODE-3951 and seeing if they run and pass now (both tests in this file, and the session poc one)?